### PR TITLE
Upgrade to Hono 2.0.0

### PR DESCRIPTION
**Pull request by sophokles73**
This is for #375

The changes in this PR will allow the chart to be used to install Hono versions >= 2.0.0. At the same time it will no longer be possible to use the chart to install Hono 1.x.

One important goal of the changes is to prepare the chart for being able to replace the shell script based creation of the example keys with a cert-manager approach without requiring the chart's major version to be increased again.

### CODE REVIEW

**Comment by sophokles73** [comment]
@calohmn would you mind taking a look?

File: charts/hono/README.md [hunk 1]
```diff
@@ -388,78 +495,14 @@ The Helm chart supports deployment of an example data grid which can be used for
...
-| *adapters.lora.enabled*      | `false`   | Indicates if the (experimental) LoRa WAN protocol adapter should be deployed. |
-| *adapters.mqtt.enabled*      | `true`    | Indicates if the MQTT protocol adapter should be deployed. |
-
-The following command will deploy the LoRa adapter along with Hono's standard adapters (AMQP, HTTP and MQTT):
-
-```bash
-helm install --dependency-update --wait -n hono --set adapters.lora.enabled=true eclipse-hono eclipse-iot/hono
-```
-
 ## Jaeger Tracing
```
Comment (calohmn) [inline charts/hono/README.md:506]: I think everything below the `kubectl get service` command in this chapter can be removed.
Specifically, the `jaegerAgentConf` property and `SAMPLING_STRATEGIES_FILE` environment variable aren't supported any more.

The `example/jaeger/default-sampling-strategies.json` file is obsolete as well.

File: charts/hono/values.yaml [hunk 2]
```diff
@@ -1144,24 +763,12 @@ adapters:
...
-      #  keyPath: "/etc/config/key.pem"
-      #  certPath: "/etc/config/cert.pem"
-
 # authServer contains configuration properties for the Auth Server component.
 authServer:
 
   # imageName contains the name (excluding registry)
   # of the container image to use for the Auth Server
-  imageName: eclipse/hono-service-auth
+  imageName: "eclipse/hono-service-auth-native"
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply charts/hono/values.yaml:771]: Good point. Then let's simply use the JVM mode image instead by default.

[the remark this answers is not in the document]
Comment (calohmn) [reply charts/hono/values.yaml:771]: > I think there should at least be a corresponding note in the `Using specific Container Images` chapter of the README.

A note that either the non-custom, default auth-server image version has to be used (`--set authServer.imageTag=2.0.0`) or, if the non-native custom auth-server image shall be used, the resources have to be adapted as well:
```
authServer:
  imageName: "eclipse/hono-service-auth"
  resources:
    requests:
      memory: "196Mi"
    limits:
      memory: "256Mi"
```

File: charts/hono/README.md [hunk 3]
```diff
@@ -29,17 +29,18 @@ The instructions below illustrate how Hono can be installed to the `hono` name s
...
+   ```bash
+   kubectl create namespace hono
+   ```
+1. Download and extract the chart to folder `hono` in the current working directory:
+   ```bash
+   helm pull --untar eclipse-iot/hono
+   ```
+1. Install the chart to name space `hono` using release name `eclipse-hono`:
+   ```bash
+   helm install --dependency-update --wait -n hono eclipse-hono hono
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply charts/hono/README.md:42]: In order for the `--dependency-update` flag to have any effect when doing `helm install`, the chart needs to be installed from a local folder. Alternatively, we could omit the `--dependency-update` flag from the example command line(s) for installing, given that we are usually talking about *first-time* installation anyway ...
WDYT?

File: charts/hono/values.yaml [hunk 4]
```diff
@@ -1672,148 +1238,15 @@ mongodb:
...
   # do not use the default name ("mongodb") because it will lead to a config
   # property name clash in the device registry pod when using "hono" as the release name
-  nameOverride: monogodb-server
-
-# deviceConnectionService contains configuration properties for the
-# Device Connection service.
-# Note that the Device Connection service will only be used if "useCommandRouter" is false.
-# DEPRECATED This property will be removed when this chart gets updated to support Hono 2.0.0.
-# Please use the Command Router service instead.
-deviceConnectionService:
```
[the remark this answers is not in the document]
Comment (calohmn) [reply charts/hono/values.yaml:1682]: There is one place left mentioning the `deviceConnectionSpec` in the README.

File: charts/hono/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (calohmn) [reply charts/hono/README.md:42]: Yes, I would prefer omitting the `--dependency-update` flag then.

File: charts/hono/example/ca_opts [hunk 5]
```diff
@@ -119,13 +119,6 @@ keyUsage             = keyAgreement,keyEncipherment,digitalSignature
 extendedKeyUsage     = serverAuth, clientAuth
 subjectAltName       = DNS.1:hono-adapter-amqp-vertx,DNS.2:hono-adapter-amqp-vertx.hono,DNS.3:localhost
```
[the remark this answers is not in the document]
Comment (calohmn) [reply charts/hono/example/ca_opts]: The `[ req_ext_device-connection ]` block can be removed.

Re-creation of the certificates is still missing. The `device-connection` and `kura-adapter` certificates can then be deleted.

File: charts/hono/templates/NOTES.txt [hunk 6]
```diff
@@ -13,9 +13,21 @@ You can check the current status by issuing the following command:
 The output should look similar to this:
```
Comment (calohmn) [inline charts/hono/templates/NOTES.txt:13]: The NOTES.txt still contains a link to https://eclipse.org/hono/getting-started-kafka/. That page doesn't exist any more.

And, while at it, you could also replace `service-device-registry-0` with something like `service-device-registry-64d645dd9c-q9c5q` below (no statefulset being used for device registry component).

Comment (sophokles73) [reply charts/hono/templates/NOTES.txt:13]: That depends on which type of registry you are using. The default (embedded) registry does indeed use a StatefulSet ...
But I can use a name reflecting the type of registry being installed ...

Comment (calohmn) [reply charts/hono/templates/NOTES.txt:13]: Oh, right, I only checked with the mongoDB and jdbc registries.

**Comment by sophokles73** [comment]
@calohmn would you mind taking another look?

**Review by calohmn — APPROVED** [review]
LGTM

**Comment by sophokles73** [comment]
@calohmn do you think we should squash all changes into a single commit?

**Comment by calohmn** [comment]
@sophokles73 I think it's good having the separate commits.

**Pull request merged**